### PR TITLE
Reservations RP 2018-06-01 Preview: Add RedHat in reserved resource t…

### PR DESCRIPTION
**Pull request by luluRagdoll**
…ype enum

## Description

**LINK:** 
https://github.com/Azure/azure-rest-api-specs/pull/4370
---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by dsgouda** [comment]
Please fix build failures

File: src/SDKs/Reservations/Management.Reservations/Generated/SdkInfo_AzureReservationAPI.cs [hunk 1]
```diff
@@ -36,5 +36,5 @@ internal static partial class SdkInfo
       public static readonly String CodeGenerationErrors = "";
       public static readonly String GithubRepoName = "azure-rest-api-specs";
       // END: Code Generation Metadata Section  
-	  }
```
Comment (dsgouda) [inline src/SDKs/Reservations/Management.Reservations/Generated/SdkInfo_AzureReservationAPI.cs]: This file should not be edited manually. Please regenerate the code. 
If there are issues with the generated file, please do the following
1. Pull latest changes from psSdkJson6
2. Run 
```
git clean -xdff
msbuild build.proj
```
3. Regenerate the code

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/Reservations/Management.Reservations/Generated/SdkInfo_AzureReservationAPI.cs (hunk 1 above)
Comment (luluRagdoll) [reply src/SDKs/Reservations/Management.Reservations/Generated/SdkInfo_AzureReservationAPI.cs]: @dsgouda  Hi, 
I remembered this is a bug in the auto-generating script? Last time I had the same issue and had to fix this manually. :) 

Last time my PR : 
https://github.com/Azure/azure-sdk-for-net/pull/4725

Comment (dsgouda) [reply src/SDKs/Reservations/Management.Reservations/Generated/SdkInfo_AzureReservationAPI.cs]: @luluRagdoll This has been fixed in our latest build tools. please follow the steps I mentioned above and regenerate the code

Comment (dsgouda) [reply src/SDKs/Reservations/Management.Reservations/Generated/SdkInfo_AzureReservationAPI.cs]: @luluRagdoll 
Please follow the steps described above and regenerate the code

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please update the version number and PackageReleaseNotes in csproj

File: src/SDKs/Reservations/Management.Reservations/Generated/SdkInfo_AzureReservationAPI.cs (hunk 1 above)
Comment (dsgouda) [reply src/SDKs/Reservations/Management.Reservations/Generated/SdkInfo_AzureReservationAPI.cs]: @luluRagdoll Please follow the steps described above and regenerate the code

**Comment by dsgouda** [comment]
@luluRagdoll PR has been inactive since more than 2 weeks. Please open a new one once you've addressed the comments

**Pull request closed without merging**